docs/content/design/communication.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Communication"
+date: 2018-02-11T15:57:56-05:00
+---
+
+## Communication flow
+
+This is the story of a time long ago. A time of myth and legend when the ancient Gods were petty and cruel and they plagued build with irreproducibility.
+Only one project dared to challenge their power...Athens. Athens possessed a strength the world had never seen. Filling its cache.
+
+### Clean plate
+
+At the beginning, there's theoretical state when a cache of proxy and registry is empty.
+
+When User makes a request at this ancient time, it works as described on the flow below.
+
+- User contacts proxy asking for module M in version v1.0
+- Proxy checks whether or not it has this module in its storage. It has not.
+-So it responds with a redirect to Olympus and schedules its own job to check with Olympus in few units of time, giving Olympus space to fill its cache.
```

Comment:
What do you think about only saying "... schedules a background job to fetch the module from Olympus ..."